shadows/framework/src/main/java/org/robolectric/shadows/ShadowLoadedApk.java

```diff
@@ -23,6 +42,50 @@ public ClassLoader getSplitClassLoader(String splitName) throws NameNotFoundExce
     return this.getClass().getClassLoader();
   }
 
+  private void tryInitAppComponentFactory(LoadedApk realLoadedApk) {
+    if (RuntimeEnvironment.getApiLevel() >= VERSION_CODES.P) {
+      ApplicationInfo applicationInfo = realLoadedApk.getApplicationInfo();
+      if (applicationInfo == null || applicationInfo.appComponentFactory == null) {
+        return;
+      }
+      _LoadedApk_ loadedApkReflector = reflector(_LoadedApk_.class, realLoadedApk);
+      if (!loadedApkReflector.getIncludeCode()) {
+        return;
+      }
+      String fullQualifiedClassName =
+          replaceLastDotWith$IfInnerStaticClass(
+              calculateFullQualifiedClassName(
+                  applicationInfo.appComponentFactory, applicationInfo.packageName));
+      android.app.AppComponentFactory factory =
+          (android.app.AppComponentFactory) newInstanceOf(fullQualifiedClassName);
+      if (factory == null) {
+        factory = new AppComponentFactory();
```

Comment:
@hoisie I think I need to remove `android.app.AppComponentFactory` import and use full qualified class name here.

Reply:
Updated, and let's check the CI result.